exp/internal/controllers/machinepool_controller_phases.go

```diff
@@ -398,19 +412,30 @@ func (r *MachinePoolReconciler) createOrUpdateMachines(ctx context.Context, mp *
 	var errs []error
 	for i := range infraMachines {
 		infraMachine := &infraMachines[i]
+
+		// Get Spec.ProviderID from the infraMachine.
+		var providerID string
+		var node corev1.Node
```

Comment:
```suggestion
		var node *corev1.Node
```

Reply:
Otherwise we will pass an empty Node object into computeDesiredMachine if providerID is not set which seems a bit awkward

Reply:
yep for sure thanks for catching